src/Units/DoRUnits.java

```diff
@@ -734,11 +734,11 @@ public GameEventQueue getTurnInitEvents(Unit self, MapMaster map)
     {
       GameEventQueue events = new GameEventQueue();
       for( Unit cargo : self.heldUnits )
+      {
+        events.add(new HealUnitEvent(cargo, self.CO.getRepairPower(), self.CO)); // Event handles cost logic
```

Comment:
Now we are healing even when they don't need it? I mean, it works I guess.

Reply:
The event's made to handle healing units that don't need healing.